src/app/guards/authorize.guard.ts

```diff
@@ -67,7 +68,14 @@ export class AuthorizeGuard implements CanActivateChild {
   }
 
   sendUserToRedirectURL(oauthSession: RequestInfoForm): Observable<boolean> {
+    // Prevent double redirection caused by races using simple semaphore like construct
+    if (this.inTheMiddleOfRedirect) {
+      return;
+    }
+
     this.window.location.href = oauthSession.redirectUrl
+    this.inTheMiddleOfRedirect = true;
```

Comment:
The problem is that we have 2 redirects, one is called by GTM library it self, and the other one is called by `sendUserToRedirectURL`. 

So in reality the `sendUserToRedirectURL` is only call once, and that is why this boolean check does not work .